src/ie9-ajax.js

```diff
@@ -51,15 +53,24 @@ function ie9Ajax(method, url, opts) {
       }
     };
 
-    function handleError() {
-      reject(new Error('There was an error with the XDR'));
+    function handleError(error) {
+      logger.error(`There was an error with the XDR\n${JSON.stringify(error, null, '  ')}`);
     }
 
     xdr.onerror = handleError;
     xdr.ontimeout = handleError;
 
-    xdr.open(method, authToUrl(url, opts));
-    xdr.send(opts.data);
+    try {
+      xdr.open(method, authToUrl(url, opts));
+      xdr.send(opts.data);
+    } catch (error) {
+      // using trim because the error message has a bunch of white space for some reason
+      if (error.message.trim() === 'Access is denied.') {
+        logger.error('Please ensure Shopify JavaScript Buy SDK is run over https');
+      } else {
+        logger.error(`There was an error when running XDR\n${error.message}`);
```

Comment:
Ideally we'd like to do something like:
```
logger.error('Please ensure Shopify JavaScript Buy SDK is ru...
reject(new Error('Please ensure Shopify JavaScript Buy SDK is ru...
```

But it's tough because `logger.error` is equivalent to `throw new Error` which means we cannot execute statements in after `logger.error`.

Advantages I see in us using `logger` is that in theory we could start logging errors people have often and be proactive in fixing those issues. The downside of the logger is that you basically get an error saying that a promise error was not handled properly with 0 line numbers :( (unless the user handles their errors properly by adding a catch statement and reviewing stack traces)

Using `reject` has similar issues where if the user isn't handling  errors and just just a dumb `catch` statement you'll have similar issues.